ryven/main/Ryven.py

```diff
@@ -1,105 +1,560 @@
+
+    # Check, if project file exists
+    if args.project:
+        project = pathlib.Path(args.project)
+        if project.exists():
+            args.project = project
+        else:
+            parser.error(
+                'project file does not exist')
+
+    # Put example into 'project' argument
+    if args.example:
+        if args.project:
+            parser.error(
+                'when loading an example, no argument PROJECT is allowed')
+
+        args.project = pathlib.Path(exampledir, args.example).with_suffix('.json')
+
+    return args, remaining_args
+
+
+def process_nodes(nodes, requested_nodes=[]):
+    """Take a list of nodes, check it and convert it to `[NodesPackage]`.
+
+    It also removes duplicates based on the name (and not the contents!).
+
+    Parameters
+    ----------
+    nodes : list of str|pathlib.Path|NodesPackage
+        A list of nodes. The node can a `NodesPackage` instance, in which case
+        the node is copied into the resulting list; otherwise the node is
+        considered as a path to 'nodes.py'. If 'nodes.py' is found in the path,
+        a `NodesPackage` instance is created and added to the resulting list.
+        If 'nodes.py' cannot be found in the path, the package is searchd in
+        Ryven's directory, e.g. if "std" is given and not found locally, the
+        "std" package included in Ryven is loaded.
+    requested_nodes : list of NodesPackage, optional
+        A list of nodes, which were requested. These take precedence over
+        `nodes`.
+        The default is `[]`.
+
+    Returns
+    -------
+    list of NodesPackage
+        unique nodes DESCRIPTION.
+
+    """
+    from ryven.main.nodes_package import NodesPackage
+
+    node_packages = set()
+    nodes_not_found = set()
+    for node in nodes:
+        if isinstance(node, NodesPackage):
+            node_packages.add(node)
+        else:
+            # For backward compatibility we have to deal with Windows and Posix
+            # paths in the project's file
+            node_windows_path = pathlib.PureWindowsPath(node)
+            node_posix_path = pathlib.PurePosixPath(node)
+            if len(node_windows_path.parts) > len(node_posix_path.parts):
+                node_path = pathlib.Path(node_windows_path)
+            else:
+                node_path = pathlib.Path(node_posix_path)
+            if node_path.joinpath('nodes.py').exists():
+                node_packages.add(NodesPackage(str(node_path)))
+            else:
+                # Try to find the node in Ryven's dirs
+                for name in ('package', 'example_nodes'):
+                    sys_node = pathlib.Path(abs_path_from_package_dir(name)).joinpath(node)
+                    if sys_node.joinpath('nodes.py').exists():
+                        node_packages.add(NodesPackage(str(sys_node)))
+                        break
+                else:
+                    nodes_not_found.add(node_path)
```

Comment:
not sure about this. which directory does `'package'` refer to? also, it doesn't seem to search in custom nodes (`ryven_dir_path()/nodes/`, aka `~/.ryven/nodes/` on Linux). I think it should be like this:

```python
[...]
                # Try to find the node in Ryven's dirs
                example_nodes_pkg_path = pathlib.Path(abs_path_from_package_dir('example_nodes')).joinpath(node)
                if example_nodes_pkg_path.exists():
                    node_packages.add(NodesPackage(str(example_nodes_pkg_path)))
                    continue
                
                custom_nodes_pkg_path = pathlib.Path(ryven_dir_path()).joinpath('nodes').joinpath(node)
                if custom_nodes_pkg_path.exists():
                    node_packages.add(NodesPackage(str(custom_nodes_pkg_path)))
                    continue
                
                else:
                    nodes_not_found.add(node_path)
[...]
```
do you agree? I can push these changes in that case. seems to work for me.

Reply:
Yes, you are absolutely right. It's just that I don't have this (recent) change on my radar yet.

I'll change it myself (if that's ok with you), because I still have one or two other changes in the pipeline.

Reply:
sure, if you can push them soon please go ahead so I can add some of my tiny changes as well